contracts/reserve/ExchangeHelper.sol

```diff
@@ -301,35 +301,37 @@ contract ExchangeHelper is DAOUpgradeableContract {
 			nameService.getAddress("UNISWAP_ROUTER")
 		);
 		address[] memory path = new address[](2);
-
+		address wETH = uniswapContract.WETH();
 		uint256[] memory swap;
-		if (address(_token) == address(0x0)) {
-			if (_isBuy) {
-				path[0] = uniswapContract.WETH();
-				path[1] = daiAddress;
-				swap = uniswapContract.swapExactETHForTokens{
-					value: _tokenAmount
-				}(_minDAIAmount, path, address(this), block.timestamp);
-				return swap;
-			} else {
-				path[0] = daiAddress;
-				path[1] = uniswapContract.WETH();
-				swap = uniswapContract.swapExactTokensForETH(
-					_tokenAmount,
-					_minTokenReturn,
-					path,
-					_receiver,
-					block.timestamp
-				);
-				return swap;
-			}
+		path[0] = _inputToken == address(0x0) ? wETH : _inputToken;
+		path[1] = _outputToken == address(0x0) ? wETH : _outputToken;
+
+		if (path[0] == wETH) {
+			swap = uniswapContract.swapExactETHForTokens{ value: _tokenAmount }(
+				_minDAIAmount,
+				path,
+				address(this),
+				block.timestamp
+			);
+			return swap;
+		} else if (path[1] == wETH) {
+			swap = uniswapContract.swapExactTokensForETH(
+				_tokenAmount,
+				_minTokenReturn,
+				path,
+				_receiver,
+				block.timestamp
+			);
+			return swap;
 		} else {
-			if (_isBuy) _token.approve(address(uniswapContract), _tokenAmount);
-			path[0] = _isBuy ? address(_token) : daiAddress;
-			path[1] = _isBuy ? daiAddress : address(_token);
+			if (path[1] == daiAddress)
```

Comment:
add a comment explaining this

Reply:
and also make a local variable isBuy = path[1] == daiAddress

Reply:
changed and added comment